lib/ansible/modules/cloud/ovirt/ovirt_snapshots.py

```diff
@@ -0,0 +1,258 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2016 Red Hat, Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import traceback
+
+try:
+    import ovirtsdk4.types as otypes
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ovirt import (
+    check_sdk,
+    create_connection,
+    get_dict_of_struct,
+    get_entity,
+    ovirt_full_argument_spec,
+    search_by_name,
+    wait,
+)
+
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: ovirt_snapshots
+short_description: "Module to manage Virtual Machine Snapshots in oVirt"
+version_added: "2.3"
+author: "Ondra Machacek (@machacekondra)"
+description:
+    - "Module to manage Virtual Machine Snapshots in oVirt"
+options:
+    snapshot_id:
+        description:
+            - "ID of the snapshot to manage."
+    vm_name:
+        description:
+            - "Name of the Virtual Machine to manage."
+        required: true
+    state:
+        description:
+            - "Should the Virtual Machine snapshot be restore/present/absent."
+        choices: ['restore', 'present', 'absent']
+        default: present
+    description:
+        description:
+            - "Description of the snapshot."
+    use_memory:
+        description:
+            - "If I(true) and C(state) is I(present) save memory of the Virtual
+               Machine if it's running."
+            - "If I(true) and C(state) is I(restore) restore memory of the
+               Virtual Machine."
+            - "Note that Virtual Machine will be paused while saving the memory."
+notes:
+    - "Note that without a guest agent the data on the created snapshot may be
+       inconsistent."
+    - "Deleting a snapshot does not remove any information from the virtual
+       machine - it simply removes a return-point. However, restoring a virtual
+       machine from a snapshot deletes any content that was written to the
+       virtual machine after the time the snapshot was taken."
+extends_documentation_fragment: ovirt
+'''
+
+
+EXAMPLES = '''
```

Comment:
In addition to examples, can you add docs on `RETURN` like what's in the `snapshot_facts` module below?